rust/sbp/src/lib.rs

```diff
@@ -10,12 +10,66 @@ pub mod serialize;
 #[cfg(feature = "sbp2json")]
 pub mod sbp2json;
 
+use std::convert::{From, Into};
 use std::error;
 use std::fmt;
 use std::result;
 
+#[cfg(feature = "sbp_serde")]
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
 pub type Result<T> = result::Result<T, Error>;
 
+pub const SBP_MAX_PAYLOAD: usize = 256;
+
+#[derive(Debug, Clone)]
+pub struct SbpString(Vec<u8>);
+
+#[cfg(feature = "sbp_serde")]
+impl Serialize for SbpString {
+    fn serialize<S>(&self, serializer: S) -> result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let s: String = self.clone().into();
+        serializer.serialize_str(&s)
+    }
+}
+
+#[cfg(feature = "sbp_serde")]
+impl<'de> Deserialize<'de> for SbpString {
+    fn deserialize<D>(deserializer: D) -> result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Deserialize::deserialize(deserializer).map(|s: String| SbpString::from(s))
+    }
+}
+
+impl From<String> for SbpString {
+    fn from(s: String) -> SbpString {
+        SbpString(s.as_bytes().to_vec())
+    }
+}
+
+impl Into<String> for SbpString {
+    fn into(self) -> String {
+        String::from_utf8_lossy(&self.0).into()
+    }
+}
```

Comment:
It looks like the orphaning rules changed in v1.41 to allow to `impl From<SbpString> for String`: https://doc.rust-lang.org/stable/std/convert/trait.Into.html#implementing-into-for-conversions-to-external-types-in-old-versions-of-rust

I'm not sure if we want to restrict the version of rust we require.

Reply:
I don't many thoughts on this at the moment, but 1.41 feels too new since it came out this year-- despite that it seems like it's supported on all recent Ubuntu releases: https://packages.ubuntu.com/search?keywords=rustc (as a security update though)